### PR TITLE
Re-implement the backup binlog API with gRPC.

### PR DESCRIPTION
**Pull request by kfyharukz**
Signed-off-by: Kazuhito MATSUDA <kazuhito.matuda@gmail.com>

### CODE REVIEW

File: server/backup_binlog_grpc.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	}
+	if awsErr.StatusCode() != http.StatusNotFound {
+		s.agent.sem.Release(1)
+		log.Error("failed to get objects", map[string]interface{}{
+			log.FnError: err,
+		})
+		return nil, status.Errorf(codes.Internal, "failed to get object: err=%+v", err)
+	}
+
+	// TODO: change user
```
Comment (masa213f) [inline server/backup_binlog_grpc.go:80]: Could you add an issue? I'm afraid to forget to update here.

File: server/backup_binlog_grpc_test.go [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+		test_utils.StopMinIO(agentTestPrefix + "minio")
+		os.RemoveAll(tmpDir)
+		os.RemoveAll(binlogDir)
+	})
+
+	It("should flush and backup binlog", func() {
+		By("calling /flush-backup-binlog API")
+		req := &proto.FlushAndBackupBinlogRequest{
+			Token:           token,
+			BackupId:        binlogPrefix,
```
Comment (masa213f) [inline server/backup_binlog_grpc_test.go:103]: Could you define another variable like `backupID := "test-backup-id"`.
`binlogPrefix` is confusing.

File: server/backup_binlog_grpc_test.go [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+		}
+
+		By("checking the uploaded binlog file is deleted")
+		binlogName := binlogDir + "/" + binlogPrefix + ".000001"
+		_, err = os.Stat(binlogName)
+		Expect(os.IsNotExist(err)).Should(BeTrue())
+
+		By("calling /flush-backup-binlog API with the same prefix")
```
Comment (masa213f) [inline server/backup_binlog_grpc_test.go:148]:
```suggestion
		By("calling /flush-backup-binlog API with the same backup ID")
```

File: server/backup_binlog_grpc.go (hunk 1 above)
Comment (kfyharukz) [reply server/backup_binlog_grpc.go:80]: Done.
https://github.com/cybozu-go/moco/issues/208

**Review by masa213f — APPROVED** [review]
LGTM

**Pull request merged**